File: src/main/java/每日一题/ClimbStairs.java
package 每日一题;

public class ClimbStairs {
    public int climbStairs(int n) {
        int[] a  = new int[n];
        a[0] = 0;
        a[1] = 1;
        a[2] = 2;
        for(int i=3;i<=n;i++){
            a[i] = a[i-1]+a[i-2];
        }
        return a[n];
    }
}

File: src/main/java/五毒/第一周/第三遍/climb_stairs.go
package main

import "fmt"

func climbStairs(n int) int {
	var cur int = 2
	var prev int = 1
	if n <= 2 {
		return n
	}
	for i := 3; i <= n; i++ {
		var temp int = cur
		cur = cur + prev
		prev = temp
	}
	return cur
}

func main() {
	fmt.Println(climbStairs(4))
}
